File: Rgx.js
class Rgx {

    constructor(text = "texto"){
        this.text = text
    }

    match (regex = /x/) {
        console.log(this.text.match(regex))
    }

    replace (regex = /x/, another) {
        console.log(this.text.replace(regex, another))
    }

}

module.exports = Rgx

File: Rgx.rb
module Utils

    class Rgx
        def initialize (text = "Texto")
            @text = text
        end

        def match(regex = /x/) 
            puts @text.scan(regex)
            puts "--------------------------"
        end

        def replace(regex = /x/, another) 
            puts @text.gsub! regex, another
            puts "--------------------------"
        end
    end

end